src/renderers/gl/renderer.cpp

```diff
@@ -377,24 +377,30 @@ auto mrg::Renderer::render(mg::RenderableList const& renderables) const -> std::
 
 void mrg::Renderer::draw(mg::Renderable const& renderable) const
 {
+    auto const texture = gl_interface->as_texture(renderable.buffer());
     auto const clip_area = renderable.clip_area();
     if (clip_area)
     {
         glEnable(GL_SCISSOR_TEST);
-        glScissor(
-            clip_area.value().top_left.x.as_int() -
-                viewport.top_left.x.as_int(),
+
+        auto const invert_clip_area = (output_surface->layout() == graphics::gl::OutputSurface::Layout::TopRowFirst) ==
+            (texture->layout() == mg::gl::Texture::Layout::TopRowFirst);
```

Comment:
Is this correct? I expected:
```suggestion
        auto const invert_clip_area = (output_surface->layout() == graphics::gl::OutputSurface::Layout::TopRowFirst) !=
            (texture->layout() == mg::gl::Texture::Layout::TopRowFirst);
```

Reply:
That is definitely confusing when the PR is titled "fixing clip area when an output is inverted but the surface is not" and the comment is "When the layout of the texture disagrees with the layout of the output surface, then the clip area needs to be modified".

Maybe clarify that text and add a comment explaining this condition?

Reply:
Done! Let me know if it sounds good enough

Reply:
I see no changes to the PR text, just a comment explaining the condition.

Reply:
Updated :+1:

Reply:
I will fix up the commits too

Reply:
Done